File: lib/src/main/java/com/sha/formvalidator/BaseFormValidator.java
package com.sha.formvalidator;

import java.util.Arrays;
import java.util.List;

public class BaseFormValidator<T extends Validatable> {

    private List<T> fields;

    public BaseFormValidator(List<T> fields) {
        this.fields = fields;
    }

    @SafeVarargs
    public BaseFormValidator(T... fields) {
        this.fields = Arrays.asList(fields);
    }

    protected boolean isValidForm(){
        boolean allValid = true;

        for (T field : fields)
            allValid = field.validate() && allValid;

        return allValid;
    }



}

File: lib/src/main/java/com/sha/formvalidator/BaseFormValidator.kt
package com.sha.formvalidator

open class BaseFormValidator<T : Validatable> {

    private var fields: List<T> = emptyList()
    val isValid: Boolean
        get() {
            var isValid = true
            fields.forEach { isValid = it.validate() && isValid }
            return isValid
        }

    constructor(fields: List<T>) {
        this.fields = fields
    }

    @SafeVarargs
    constructor(vararg fields: T) {
        this.fields = fields.asList()
    }


}
